src/faebryk/libs/kicad/fileformats.py

```diff
@@ -748,27 +824,23 @@ class C_net:
             number: int = field(**sexp_field(positional=True))
             name: str = field(**sexp_field(positional=True))
 
-        @dataclass
+        @dataclass(kw_only=True)
         class C_pcb_footprint(C_footprint):
-            @dataclass
-            class C_pad(C_footprint.C_pad):
-                net: tuple[int, str] = field(kw_only=True)
-                uuid: UUID = field(kw_only=True)
-
-            at: C_xyr
-            uuid: UUID
-            pads: list[C_pad] = field(**sexp_field(multidict=True))
+            ...
+            # TODO: kicad does not like that 'at' and 'uuid' are defined at
+            # the end of the sexp section. this happens because C_pcb_footprint is
+            # inherriting C_footprint and is thus initiated after C_footprint.
```

Comment:
maybe use

```python
def __init__(...):
    super().__init__(...)
 ```
to first initialize the `C_pcb_footprint` then the `C_footprint` class?

Reply:
This is what we want to be in here 
```python
class C_pcb_footprint(C_footprint):
        at: C_xyr # defined before the `properties` in the KiCad file (this is a must)
        uuid: UUID # defined before the `properties` in the KiCad file (this might be a must)
        pads: list[C_pad] = field(**sexp_field(multidict=True)) # defined after the `properties` etc in the KiCad file  (this might be a must)
```

e.g.

```sexp
(footprint "lcsc:C0402"
		(layer "F.Cu")
		(uuid "0420be79-ae67-43c0-9923-6025419448a6")
		(at 1.15 5 180)
		(property "Reference" "C1"
			(at 0 -4 180)
			(layer "F.SilkS")
			(uuid "a454701a-4ae5-483d-b5d7-f5567c13598a")
			(effects
				(font
					(size 0.5 0.5)
					(thickness 0.1)
				)
			)
		)
        ...
		(attr smd)
		(fp_line
			(start 1.17 0.35)
			(end 1.17 -0.35)
			(stroke
				(width 0.15)
				(type solid)
			)
			(layer "F.SilkS")
			(uuid "d3c58013-a41d-4d0b-82a0-4ad4a52cfb1b")
		)
        ...
		(pad "1" smd rect
			(at -0.55 0 180)
			(size 0.79 0.54)
			(layers "F.Cu" "F.Paste" "F.Mask")
			(net 5 "3V3_PERIPHERAL")
			(uuid "ed040bd2-787d-4bd7-adf3-f11c34eaa1df")
		)
        ...
		(model "${KIPRJMOD}/../libs/3dmodels/lcsc.3dshapes/C0402_L1.0-W0.5-H0.5.wrl"
			(offset
				(xyz 0 0 0)
			)
			(scale
				(xyz 1 1 1)
			)
			(rotate
				(xyz 0 0 0)
			)
		)
	)
```

Reply:
Fixed